scratch/stan_chol_exmpl.R

```diff
@@ -0,0 +1,200 @@
+library(tidyverse)
+library(rstan)
+
+
+
+stan_model_code_non_cent <- "
+functions {
+  matrix exponential_decay_corr_func(matrix dist_matrix, real phi, real l) {
+    int k = rows(dist_matrix);
+    matrix[k,k] corr_matrx = exp(-pow(dist_matrix / phi, l));
+    return corr_matrx;
+  }
+}
+
+data {
+  int<lower=1> n_sites;
+  int<lower=1> n_times;
+  matrix[n_sites,n_sites] dist_matrix;
+  real<lower=0> l;
+  matrix[n_sites, n_times] obs_data;
+}
+
+parameters {
+  matrix[n_sites,n_times] alpha;
+  real<lower=0> phi;
+  real<lower=0> sigma_eps;
+  //real<lower=0> sigma_obs; Can get rid of this and assume iid normal
+}
+
+transformed parameters {
+  matrix[n_sites,n_sites] Sigma;
+  matrix[n_sites,n_times] epsilon;
+
+  Sigma = sigma_eps^2 * exponential_decay_corr_func(dist_matrix, phi, l);
+
+  for (i in 1:n_times) {
+    epsilon[,i] = sigma_eps * cholesky_decompose(Sigma) * alpha[, i];
+  }
+}
+
+model {
+  //Priors
+  to_vector(alpha) ~ std_normal();
+  phi ~ uniform(0, 50);
+  sigma_eps ~ gamma(sqrt(0.1),1);
+
+  // Likelihood
+  // Assumes normally distributed observation error.
+  // obs_data are your vectors of draws the MVN at each time point (column
+  // bound to produce a matrix). The epsilon matrix are the
+  // mean of the expected realizations of the MVN draws.
+  for (i in 1:n_times) {
+    obs_data[,i] ~ normal(epsilon[,i], 1);
+  }
+}
+"
+
+
+stan_model_code_cent <- "
+functions {
+  matrix exponential_decay_corr_func(matrix dist_matrix, real phi, real l) {
+    int k = rows(dist_matrix);
+    matrix[k,k] corr_matrx = exp(-pow(dist_matrix / phi, l));
+    return corr_matrx;
+  }
+}
+
+data {
+  int<lower=1> n_sites;
+  int<lower=1> n_times;
+  matrix[n_sites,n_sites] dist_matrix;
+  real<lower=0> l;
+  matrix[n_sites,n_times] obs_data;
+}
+
+parameters {
+  real<lower=0> phi;
+  real<lower=0> sigma_eps;
+  matrix[n_sites, n_times] epsilon;
```

Comment:
In the centered parameterization, we don't have to estimate an expected value of epsilon